File: hw/fuentes/U3/HW17Maze/MazeAppPY/ec/edu/espe/Maze/model/Room.py
from model.Wall import Wall

class Room:
    def __init__(self, x, y):
        self.x = x
        self.y = y
        self.walls = []
        self.door = None

    def has_wall(self, direction):
        return any(w.get_direction() == direction and w.exists_wall() for w in self.walls)

    def has_door(self):
        return self.door is not None

    def is_corner(self, n, m):
        return (self.x == 0 or self.x == n - 1) and (self.y == 0 or self.y == m - 1)

    def set_door(self, door):
        self.door = door

    def get_walls(self):
        return self.walls

    def get_door(self):
        return self.door

    def get_x(self):
        return self.x

    def get_y(self):
        return self.y

    def add_wall(self, wall):
        if len(self.walls) < 4:
            self.walls.append(wall)

File: hw/fuentes/U3/HW17Maze/NewMazeGenerator/src/main/java/ec/edu/espe/mazegenerator/model/Room.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.mazegenerator.model;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author jorge
 */
public class Room {
    private int x;
    private int y;
    private List<Wall> walls;
    private Door door;

    public Room(int x, int y) {
        this.x = x;
        this.y = y;
        this.walls = new ArrayList<>();
    }

    public boolean hasWall(char direction) {
        return walls.stream().anyMatch(w -> w.getDirection() == direction && w.exists());
    }

    public boolean hasDoor() {
        return door != null;
    }

    public boolean isCorner() {
        return (x == 0 || x == Maze.N - 1) && (y == 0 || y == Maze.M - 1);
    }

    public void setDoor(Door door) {
        this.door = door;
    }

    public List<Wall> getWalls() {
        return walls;
    }

    public Door getDoor() {
        return door;
    }

    public int getX(){ 
        return x; 
    }
    public int getY(){ 
        return y; 
    }
    
    public void addWall(Wall wall) {
    if (walls.size() < 4) {
        walls.add(wall);
    }
}
    
}
